src/ophys_etl/modules/segmentation/graph_utils/feature_vector_segmentation.py

```diff
@@ -0,0 +1,630 @@
+
+def _get_roi(seed_obj: dict,
+             video_data: np.ndarray,
+             filter_fraction: float,
+             pixel_ignore: np.ndarray,
+             output_dict: multiprocessing.managers.DictProxy,
+             roi_id: int,
+             roi_class: type) -> None:
+    """
+    Worker method started as a multiprocessing.Process
+    to actually grow an ROI.
+
+    Parameters
+    ----------
+    seed_obj: dict
+        A dict as returned by find_peaks containing the seed
+        information for one ROI
+
+    video_data: np.ndarray
+        The subset of a video to be searched for the ROI. Shape
+        is (n_time, n_rows, n_cols) where n_rows and n_cols
+        are equal to seed_obj['rows'][1]-seed_obj['rows'][0]
+        and seed_boj['cols'][1]-seed_obj['cols'][0] (i.e. the
+        field of view has already been clipped)
+
+    filter_fraction: float
+        The fraction of brightest timesteps to use in feature
+        calculation
+
+    pixel_ignore: np.ndarray
+        A (n_rows, n_cols) array of booleans marked True at
+        any pixels that should be ignored, presumably because
+        they have already been added to an ROI
+
+    output_dict: multiprocessing.managers.DictProxy
+        The dict where the final ROI mask from this search will
+        be stored. After running this method,
+        output_dict[roi_id] will be a tuple containing the
+        origin of the field of view (i.e. the coordinates of the
+        upper left corner) and the array of booleans representing
+        the ROI's mask.
+
+    roi_id: int
+        The unique ID of this ROI. This will be used as the key
+        in output_dict for this ROI's mask
+
+    roi_class: type
+        The sub-class of PotentialROI that will be used to find
+        this ROI
+
+    Returns
+    -------
+    None
+        Results are stored in output_dict
+    """
+
+    seed_pt = seed_obj['center']
+    origin = (seed_obj['rows'][0], seed_obj['cols'][0])
+
+    roi = roi_class(seed_pt,
+                    origin,
+                    video_data,
+                    filter_fraction,
+                    pixel_ignore=pixel_ignore)
+
+    final_mask = roi.get_mask()
+    output_dict[roi_id] = (origin, final_mask)
+    return None
+
+
+def convert_to_lims_roi(origin: Tuple[int, int],
+                        mask: np.ndarray,
+                        roi_id: int = 0) -> ExtractROI:
+    """
+    Convert an origin and a pixel mask into a LIMS-friendly
+    JSONized ROI
+
+    Parameters
+    ----------
+    origin: Tuple[int, int]
+        The global coordinates of the upper right corner of the pixel mask
+
+    mask: np.ndarray
+        A 2D array of booleans marked as True at the ROI's pixels
+
+    roi_id: int
+        default: 0
+
+    Returns
+    --------
+    roi: dict
+        an ExtractROI matching the input data
+    """
+    # trim mask
+    row0 = 0
+    col0 = 0
+    row1 = mask.shape[0]
+    col1 = mask.shape[1]
+    for i_row in range(mask.shape[0]):
```

Comment:
it seems like these 4 for loops could be simpler, like:
```
nz_coords = np.argwhere(mask > 0)
row0 = nz_coords[:, 0].min()
...
```

Reply:
Done